UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/AuthTokenProvider.cs

```diff
@@ -59,26 +65,23 @@ private async Task<AccessTokenItem> GetNewAuthToken(string resource)
 
         private void AddAuthTokenToCache(string key, AccessTokenItem accessTokenItem)
         {
-            var tokenExpiryMinutes = accessTokenItem.ExpiresIn.Subtract(DateTime.UtcNow).TotalMinutes;
-            var deductTokenExpiryMinutes = _essManagedIdentityConfiguration.Value.DeductTokenExpiryMinutes < tokenExpiryMinutes ? _essManagedIdentityConfiguration.Value.DeductTokenExpiryMinutes : 1;
-            var options = new DistributedCacheEntryOptions().SetSlidingExpiration(TimeSpan.FromMinutes(tokenExpiryMinutes - deductTokenExpiryMinutes));
+            _logger.LogInformation(EventIds.CachingExternalEndPointTokenStarted.ToEventId(), "Adding new access token in cache to call external endpoint | {DateTime} | _X-Correlation-ID:{CorrelationId}", DateTime.Now.ToUniversalTime(), CommonHelper.CorrelationID);
+
+            double tokenExpiryMinutes = accessTokenItem.ExpiresIn.Subtract(DateTime.UtcNow).TotalMinutes;
+            double deductTokenExpiryMinutes = _essManagedIdentityConfiguration.Value.DeductTokenExpiryMinutes < tokenExpiryMinutes ? _essManagedIdentityConfiguration.Value.DeductTokenExpiryMinutes : 1;
+            DistributedCacheEntryOptions? options = new DistributedCacheEntryOptions().SetSlidingExpiration(TimeSpan.FromMinutes(tokenExpiryMinutes - deductTokenExpiryMinutes));
```

Comment:
We are using SlidingExpiration to configure the token's expiration to occur one minute before the real expiration time.
This resolves the situation because it is likely that the new access token will be issued before the expiration time.

SlidingExpiration:- will expire the entry if it hasn't been accessed in a set amount of time.
AbsoluteExpiration:- will expire the entry after a set amount of time.

Reply:
Resolved as suggested